core/db/migrate/20220113052823_create_payment_sources.rb

```diff
@@ -0,0 +1,21 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_payment_profile_id, index: { unique: true }
+      t.string :type, index: true
+
+      t.references :payment_method, index: true, foreign_key: { to_table: :spree_payment_methods }
+      t.references :user, index: true, foreign_key: { to_table: :spree_users }
```

Comment:
@GrzegorzStochel I think we should skip the foreign key here as the user is optional, right?

Reply:
I think the purpose of the foreign key is to raise the error while we try to delete parents if there are children records in the DB. Does `optional` change anything here?

Reply:
You are correct, my bad!